apis/network/v1alpha3/resourcerecordset_types.go

```diff
@@ -0,0 +1,149 @@
+)
+
+// +kubebuilder:object:root=true
+
+// ResourceRecordSet is a managed resource that represents an AWS Route53 Resource Record.
+// +kubebuilder:printcolumn:name="NAME",type="string",JSONPath=".status.forProvider.name"
+// +kubebuilder:printcolumn:name="TYPE",type="string",JSONPath=".status.atProvider.Type"
+// +kubebuilder:printcolumn:name="READY",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].status"
+// +kubebuilder:printcolumn:name="SYNCED",type="string",JSONPath=".status.conditions[?(@.type=='Synced')].status"
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+type ResourceRecordSet struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ResourceRecordSetSpec   `json:"spec"`
+	Status ResourceRecordSetStatus `json:"status,omitempty"`
+}
+
+// ResourceRecordSetSpec defines the desired state of an AWS Route53 Resource Record.
+type ResourceRecordSetSpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+	ForProvider                  ResourceRecordSetParameters `json:"forProvider"`
+}
+
+// ResourceRecordSetStatus represents the observed state of a ResourceRecordSet.
+type ResourceRecordSetStatus struct {
+	runtimev1alpha1.ResourceStatus `json:",inline"`
+	AtProvider                     ResourceRecordSetObservation `json:"atProvider"`
+}
+
+// ResourceRecord holds the DNS value to be used for the record.
+type ResourceRecord struct {
+	// The current or new DNS record value, not to exceed 4,000 characters. In the
+	// case of a DELETE action, if the current value does not match the actual value,
+	// an error is returned.
+	Value *string `json:"Value"`
+}
+
+// ResourceRecordSetParameters define the desired state of an AWS Route53 Resource Record.
+type ResourceRecordSetParameters struct {
+	// Name of the record that you want to create, update, or delete.
+	Name *string `json:"name"`
+
+	// Type represents the DNS record type
+	Type *string `json:"type"`
+
+	// The resource record cache time to live (TTL), in seconds.
+	TTL *int64 `json:"ttl,omitempty"`
+
+	// ResourceRecord holds the information about the resource records to act upon.
+	Records []string `json:"records"`
+
+	// ZoneID of the HostedZone in which you want to CREATE, CHANGE, or DELETE the Resource Record.
+	ZoneID *string `json:"zoneId,omitempty"`
+
+	// ZoneIDRef references a VPC to and retrieves its vpcId
+	// +optional
+	ZoneIDRef *runtimev1alpha1.Reference `json:"zoneIdRef,omitempty"`
+
+	// ZoneIDSelector selects a reference to a Zone to and retrieves its ZoneID
+	// +optional
+	ZoneIDSelector *runtimev1alpha1.Selector `json:"zoneIdSelector,omitempty"`
+}
+
+// ResourceRecordSetObservation keeps the state for the external resource.
+type ResourceRecordSetObservation struct {
+	// Name of the record that you want to create, update, or delete.
+	Name *string `json:"Name"`
```

Comment:
I don't think we need to repeat the name in the observation - per my comment above I think it should be the `ExternalName` annotation.